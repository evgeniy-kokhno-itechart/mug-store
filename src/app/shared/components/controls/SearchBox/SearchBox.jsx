import React, { PureComponent } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { PropTypes } from 'prop-types';
import './SearchBox.scss';

class SearchBox extends PureComponent {
  state = { query: this.props.value };

  handleChange = (e) => {
    this.setState({ query: e.currentTarget.value });
  };

  handleSubmit = (e) => {
    this.props.onSearchSubmit(e, this.state.query);
  };

  render() {
    return (
      <form className="search-form" onSubmit={this.handleSubmit}>
        <input
          className="search-form__input focusable"
          type="text"
          name="query"
          placeholder="Search..."
          value={this.state.query}
          onChange={this.handleChange}
        />

        <button type="submit" className="button button--outline focusable search-form__btn">
          <FontAwesomeIcon icon="search" />
        </button>
      </form>
    );
  }
}

SearchBox.propTypes = {
  value: PropTypes.string,
  onSearchSubmit: PropTypes.func.isRequired,
};

SearchBox.defaultProps = {
  value: 'Search...',
};

export default SearchBox;
