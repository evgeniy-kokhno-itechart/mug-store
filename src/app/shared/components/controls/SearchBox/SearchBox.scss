@import "Variables";
@import "Mixins";

.search-form {
    display: flex;

    &__input {
        font: inherit;
        padding-left: 0.5rem;
        border: 1px solid $color-gray-light-2;
        border-right: none;
        width: 100%;
    }

    &__btn {
        @include respond(tab-portrait) {
        }
    }
}