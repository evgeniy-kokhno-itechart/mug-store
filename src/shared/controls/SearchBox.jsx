import React, { PureComponent } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { PropTypes } from 'prop-types';

class SearchBox extends PureComponent {
  state = { query: this.props.value };

  handleChange = (e) => {
    this.setState({ query: e.currentTarget.value });
  };

  handleSubmit = (e) => {
    this.props.onSearchSubmit(e, this.state.query);
  };

  render() {
    return (
      <form className="input-group" onSubmit={this.handleSubmit}>
        <input
          className="form-control"
          type="text"
          name="query"
          placeholder="Search..."
          value={this.state.query}
          onChange={this.handleChange}
          aria-label="Search"
        />

        <div className="input-group-append">
          <button type="submit" className="btn btn-outline-secondary">
            <FontAwesomeIcon icon="search" />
          </button>
        </div>
      </form>
    );
  }
}

SearchBox.propTypes = {
  value: PropTypes.string,
  onSearchSubmit: PropTypes.func.isRequired,
};

SearchBox.defaultProps = {
  value: 'Search...',
};

export default SearchBox;
