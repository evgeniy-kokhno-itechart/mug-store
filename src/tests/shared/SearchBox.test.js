/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SearchBox from '../../shared/controls/SearchBox';

configure({ adapter: new Adapter() });

describe('<SearchBox />', () => {
  let onSearchSubmit;

  beforeEach(() => {
    onSearchSubmit = jest.fn();
  });

  it('renders with default props with Search... text in the input', () => {
    const SearchBoxWrapper = shallow(<SearchBox onSearchSubmit={onSearchSubmit} />);
    expect(SearchBoxWrapper.debug()).toMatchSnapshot();
  });

  it('renders with value = test search in the input', () => {
    const SearchBoxWrapper = shallow(<SearchBox value="test search" onSearchSubmit={onSearchSubmit} />);
    // SearchBoxWrapper.setState({ query: 'test search' }); // this is too late. value should be provided from the very beginning
    expect(SearchBoxWrapper.debug()).toMatchSnapshot();
  });
});
