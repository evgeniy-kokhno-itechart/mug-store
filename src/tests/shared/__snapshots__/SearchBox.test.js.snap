// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> renders with default props with Search... text in the input 1`] = `
"<form className=\\"input-group\\" onSubmit={[Function]}>
  <input className=\\"form-control\\" type=\\"text\\" name=\\"query\\" placeholder=\\"Search...\\" value=\\"Search...\\" onChange={[Function]} aria-label=\\"Search\\" />
  <div className=\\"input-group-append\\">
    <button type=\\"submit\\" className=\\"btn btn-outline-secondary\\">
      <FontAwesomeIcon icon=\\"search\\" border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} />
    </button>
  </div>
</form>"
`;

exports[`<SearchBox /> renders with value = test search in the input 1`] = `
"<form className=\\"input-group\\" onSubmit={[Function]}>
  <input className=\\"form-control\\" type=\\"text\\" name=\\"query\\" placeholder=\\"Search...\\" value=\\"test search\\" onChange={[Function]} aria-label=\\"Search\\" />
  <div className=\\"input-group-append\\">
    <button type=\\"submit\\" className=\\"btn btn-outline-secondary\\">
      <FontAwesomeIcon icon=\\"search\\" border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} />
    </button>
  </div>
</form>"
`;
